src/plugins/vis_type_timeseries/public/request_handler.ts

```diff
@@ -47,21 +49,37 @@ export const metricsRequestHandler = async ({
 
     validateInterval(parsedTimeRange, visParams, maxBuckets);
 
-    const resp = await getCoreStart().http.post(ROUTES.VIS_DATA, {
-      body: JSON.stringify({
-        timerange: {
-          timezone,
-          ...parsedTimeRange,
+    const isCurrentSession = () =>
+      !!searchSessionId && searchSessionId === dataSearch.search.session.getSessionId();
+    const untrackSearch =
+      isCurrentSession() &&
+      dataSearch.search.session.trackSearch({
+        abort: () => {
+          // TODO: support search cancellations
         },
-        query: input?.query,
-        filters: input?.filters,
-        panels: [visParams],
-        state: uiStateObj,
-        sessionId: dataSearch.search.session.getSessionId(),
-      }),
-    });
+      });
 
-    return resp;
+    try {
+      return await getCoreStart().http.post(ROUTES.VIS_DATA, {
+        body: JSON.stringify({
+          timerange: {
+            timezone,
+            ...parsedTimeRange,
+          },
+          query: input?.query,
+          filters: input?.filters,
+          panels: [visParams],
+          state: uiStateObj,
+          sessionId: searchSessionId,
+          isRestore: isCurrentSession() ? dataSearch.search.session.isRestore() : false,
```

Comment:
nit: what if we create a separate method for that in the `data` plugin and reuse it between plugins. 

```
const getSessionParams(searchSessionId) => {
  const isCurrentSession =
    !!searchSessionId && searchSessionId === dataSearch.search.session.getSessionId();

  return isCurrentSession ? {
    searchSessionId,
    isRestore: dataSearch.search.session.isRestore(),
    isStored: dataSearch.search.session.isStored(),
  } : {
    searchSessionId,
    isRestore: false,
    isStored: false,
  }
}
```

and then just do something like: 

`session: getSessionParams(sessionId)`

Reply:
Added!